apps/explorer/src/components/ownedobjects/views/UnderlyingObjectCard.tsx

```diff
@@ -42,16 +54,20 @@ export function UnderlyingObjectCard({
     }
 
     const fieldsData = getObjectFields(data);
-    return fieldsData ? (
-        <SyntaxHighlighter
-            code={JSON.stringify(
+    // Return null if there are no fields
+    if (!fieldsData || !normalizedStruct?.fields) {
+        return null;
+    }
+    return (
+        <FieldItem
+            value={
+                // show name if it is a struct
                 typeof fieldsData?.name === 'object'
                     ? { name: fieldsData.name, value: fieldsData.value }
-                    : fieldsData?.value,
-                null,
-                2
-            )}
-            language="json"
+                    : fieldsData?.value
+            }
+            // add the struct type to the value
+            type={normalizedStruct.fields[2].type}
```

Comment:
and provide to the field the correct type, find the field in normalizedStruct.fields that have the same name as the property we are looking for